### PR TITLE
add "utils" commands

### PR DESCRIPTION
**Pull request by zcyes**
as described in comments commits.

### CODE REVIEW

**Comment by zcyes** [comment]
add test case: sleep, quit, man, launch

File: libguestfs/tests/guestfish_utils.py [hunk 1]
```diff
@@ -0,0 +1,1570 @@
...
+          <target dev='hda' bus='ide'/>\n\
+        </disk>\n\
+      </devices>\n\
+    </domain>\n\
+    " % (test_domain_name, image_path)
+    f = open(test_xml, "w")
+    f.write(xml_content)
+    f.close()
+
+    os.system("virsh define %s > /dev/null" % test_xml)
```
Comment (shi2wei3) [inline libguestfs/tests/guestfish_utils.py]: We'd better do not call virsh command directly, we can try to reuse the libvirt_vm, you can reference to guestfs_add.py

File: libguestfs/tests/guestfish_utils.py [hunk 2]
```diff
@@ -0,0 +1,1570 @@
...
+    Test command add_drive:
+    """
+    add_ref = params.get("gf_add_ref", "disk")
+    readonly = params.get("gf_add_readonly", "no")
+
+    gf = utils_test.libguestfs.GuestfishTools(params)
+
+    image_path = params.get("image_path")
+    pv_name = params.get("pv_name")
+    alloc_file = '~/../../home/test_add_drive.img'
```
Comment (shi2wei3) [inline libguestfs/tests/guestfish_utils.py]: It is a bad idea to use hardcode path especially with '~', small temp images could save to /tmp or params.get("image_path")

File: libguestfs/tests/cfg/guestfish_utils.cfg [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    lv_name = vol_file
+    image_formats = "raw"
+    image_name = utils
+    tarball_file = utils.tgz
+    image_size = "1G"
+
+    variants:
+        - utils_cmds:
+            variants:
+                - add_domain:
```
Comment (yumingfei) [inline libguestfs/tests/cfg/guestfish_utils.cfg]: Shouldn't these add-operations be put into guestfs_add or some other module?

Comment (yumingfei) [inline libguestfs/tests/cfg/guestfish_utils.cfg]: BTW, what's the principle to collect these testcases?

File: libguestfs/tests/guestfish_block_dev.py [hunk 4]
```diff
@@ -34,12 +34,9 @@ def test_blockdev_flushbufs(vm, params):
     readonly = params.get("gf_add_readonly", "no")
 
     gf = utils_test.libguestfs.GuestfishTools(params)
-    if add_ref == "disk":
-        image_path = params.get("image_path")
-        gf.add_drive_opts(image_path, readonly=readonly)
-    elif add_ref == "domain":
-        vm_name = params.get("main_vm")
-        gf.add_domain(vm_name, readonly=readonly)
+
```
Comment (yumingfei) [inline libguestfs/tests/guestfish_block_dev.py]: Do you mean these test modules should only test with images but domains?

**Comment by yumingfei** [comment]
Another problem about these patches is that you'd better rename your patches' subject to
make it more clear and useful.
We need to check them and know what we have done in future.

Generally ,we have following format:
module: the changes your have done for this module.
(less than 70 character as possible) :)

**Comment by waynesun09** [comment]
Close as no update for long time

**Pull request closed without merging**